api/src/main/java/org/apache/iceberg/io/SupportsBulkOperations.java

```diff
@@ -18,12 +18,23 @@
  */
 package org.apache.iceberg.io;
 
+import java.util.concurrent.ExecutorService;
+
 public interface SupportsBulkOperations {
   /**
    * Delete the files at the given paths.
    *
    * @param pathsToDelete The paths to delete
-   * @throws BulkDeletionFailureException in
+   * @throws BulkDeletionFailureException in case of failure to delete at least 1 file
    */
   void deleteFiles(Iterable<String> pathsToDelete) throws BulkDeletionFailureException;
+
+  /**
+   * Delete the files at the given paths with the given executor service
+   *
+   * @param pathsToDelete The paths to delete
+   * @throws BulkDeletionFailureException in case of failure to delete at least 1 file
+   */
+  void deleteFiles(Iterable<String> pathsToDelete, ExecutorService svc)
```

Comment:
Are we sure passing an executor service to `FileIO` makes sense?
Also, this logic seems reasonable for regular non-bulk IO implementations.

Reply:
yeah, can we just use `ThreadPools.getWorkerPool`?

Reply:
Good point, I missed there's already a threadpool in the fileIO implementation. For now, since it's unclear if passing it in the mixin interface makes sense (I thought it did because I was thinking a compute layer at a higher layer can determine what is the appropriate executor service to use and then pass that to the file io layer for better control, but maybe it doesn't matter). 

I've updated to us the existing executor service in S3FileIO.